job_executor/worker/steps/dataset_pseudonymizer.py

```diff
@@ -20,192 +22,131 @@ def _get_unit_types(
+    pseudonymized_table = pyarrow.Table.from_arrays(
+        arrays_to_include, input_dataset.schema.names
+    )
 
-def _pseudonymize_csv(
-    input_csv_path: Path,
-    identifier_unit_id_type: Union[str, None],
-    measure_unit_id_type: Union[str, None],
-    job_id: str,
-) -> Path:
-    if identifier_unit_id_type and not measure_unit_id_type:
-        logger.info("Pseudonymizing identifier")
-        return _pseudonymize_identifier_only(
-            input_csv_path, identifier_unit_id_type, job_id
-        )
-    elif measure_unit_id_type and not identifier_unit_id_type:
-        logger.info("Pseudonymizing measure")
-        return _pseudonymize_measure_only(
-            input_csv_path, measure_unit_id_type, job_id
-        )
-    elif identifier_unit_id_type and measure_unit_id_type:
-        logger.info("Pseudonymizing identifier and measure")
-        return _pseudonymize_identifier_and_measure(
-            input_csv_path,
-            identifier_unit_id_type,
-            measure_unit_id_type,
-            job_id,
-        )
-    else:
-        logger.info("No pseudonymization")
-        return input_csv_path
+    return pseudonymized_table
 
 
-def run(input_csv_path: Path, metadata: Metadata, job_id: str) -> Path:
+def run(input_parquet_path: Path, metadata: Metadata, job_id: str) -> Path:
     """
     Pseudonymizes the identifier column of the dataset. Requests pseudonyms
     from an external service and replaces all values in the identifier column.
     """
     try:
-        logger.info(f"Pseudonymizing data {input_csv_path}")
+        logger.info(f"Pseudonymizing data {input_parquet_path}")
         identifier_unit_type, measure_unit_type = _get_unit_types(metadata)
         identifier_unit_id_type = (
             None
             if identifier_unit_type is None
-            else microdata_validator.get_unit_id_type_for_unit_type(
+            else microdata_tools.get_unit_id_type_for_unit_type(
                 identifier_unit_type
             )
         )
         measure_unit_id_type = (
             None
             if measure_unit_type is None
-            else microdata_validator.get_unit_id_type_for_unit_type(
+            else microdata_tools.get_unit_id_type_for_unit_type(
                 measure_unit_type
             )
         )
-        output_file = _pseudonymize_csv(
-            input_csv_path,
+        pseudonymized_table = _pseudonymize(
+            input_parquet_path,
             identifier_unit_id_type,
             measure_unit_id_type,
             job_id,
         )
-        logger.info(f"Pseudonymization step done {output_file}")
-        return output_file
+        output_path = (
+            input_parquet_path.parent
+            / f"{input_parquet_path.stem}_pseudonymized.parquet"
+        )
+
+        parquet.write_table(pseudonymized_table, output_path)
+
+        logger.info(f"Pseudonymization step done {output_path}")
+        return output_path
     except Exception as e:
```

Comment:
We could catch the named exception "UnregisteredUnitTypeError" here so we could give a more specific error message to the user:
```
except UnregisteredUnitTypeError as e:
    raise BuilderStepError("Error during pseudonymization: {str(e)}") from e
```

ref: https://github.com/statisticsnorway/microdata-tools/blob/master/microdata_tools/validation/components/unit_id_types.py#L22

Reply:
But keep the general Exception catch too, to catch the unexpected stuff 👍🏻